### PR TITLE
Update tests, test if variable are defined, puppet 4 compatibility, add proxy snmp configuration

### PR DESCRIPTION
**Pull request by jnquintin**

### CODE REVIEW

File: manifests/params.pp [hunk 1]
```diff
@@ -16,272 +16,327 @@
 class snmp::params {
   # If we have a top scope variable defined, use it, otherwise fall back to a
   # hardcoded value.
-  $agentaddress = $::snmp_agentaddress ? {
-    undef   => [ 'udp:127.0.0.1:161', 'udp6:[::1]:161' ],
-    default => $::snmp_agentaddress,
+  if defined('$snmp_agentaddress') {
```
Comment (traylenator) [inline manifests/params.pp]: I've not tried this bit it looks wrong. '$::variable' is not defined as it is a literal.

It should I believe be:

```puppet
if getvar('::snmp_agentaddress') {

}

Comment (jnquintin) [inline manifests/params.pp]: You have right, i have done the correction on my pull request of the project puppet network but I forgot to back port it in this projet.

**Comment by razorsedge** [comment]
@jnquintin Thanks.  There are a lot of good things in here, but do me a favor and break things out into separate PRs.  I see four different topics: 1) adding proxy support, 2) adding AgentX sub-agent support, 3) fixing rwuser, and 4) fixing strict variables.  Please include updates to the README.markdown documenting the additions as well as class input validation and spec tests.

You also have a lot of bits to fix the failing rspec tests in .travis.yml, Gemfile, Rakefile, etc.  I have recently fixed all of that in the `develop` branch.  Those commits can be dropped.

**Comment by jnquintin** [comment]
new PR provided

**Comment by razorsedge** [comment]
Broken out into PRs #94, #95, #96, and #97.

**Pull request closed without merging**